### DISCUSSION TITLE
Custom callbacks

### DISCUSSION THREAD

**Discussion by vicovictor** [docker-mailserver/docker-mailserver#4153] [General]
Is there a way to do some callbacks before/after MTA/MDA/etc.?

Use case: Say I want to block user@gmail.com from sending me emails. I set it up on a custom web UI. So every time user@gmail.com sends me an email, it will not arrive on my inbox/spam. The system either accepts it without doing anything, or bounces it back, depending on how I program it.

In other words, I would like to be able to build some features around DMS, in the backend, without touching DMS codebase. Via callbacks, or plugins.

**Comment by polarathene** [comment]
You can look up Postfix config docs for `master.cf`, there is a way to send mail to a script I think. There is also content filter and milters, DMS doesn't have any official support for any of that, you'd need to figure it out and implement yourself.

Potentially look at other projects like Stalwart and Mailcow, perhaps they cover that?
